crates/solver/src/liquidity/zeroex.rs

```diff
@@ -79,36 +120,136 @@ impl ZeroExLiquidity {
     }
 }
 
+/// Calls the 0x API to get all orders and initializes the cache with them.
+async fn init_cache(
+    api: Arc<dyn ZeroExApi>,
+    cache: Arc<Mutex<HashMap<String, OrderRecord>>>,
+    sender: Address,
+) -> Result<()> {
+    cache.lock().await.clear(); // init can be called multiple times, so clear the cache first
+
+    let queries = &[
+        // orders fillable by anyone
+        OrdersQuery::default(),
+        // orders fillable only by our settlement contract
+        OrdersQuery {
+            sender: Some(sender),
+            ..Default::default()
+        },
+    ];
+
+    let zeroex_orders_results =
+        futures::future::join_all(queries.iter().map(|query| api.get_orders(query))).await;
+    let mut zeroex_orders = vec![];
+    for result in zeroex_orders_results {
+        match result {
+            Ok(orders) => zeroex_orders.extend(orders),
+            Err(err) => {
+                return Err(anyhow::anyhow!(
+                    "ZeroExResponse error during liqudity fetching: {}",
+                    err
+                ));
+            }
+        }
+    }
+
+    let mut cache = cache.lock().await;
+    cache.extend(
```

Comment:
I'm a bit worried that this will result in high memory usage. At some point we started to filter out many small orders because we simply had too many. Maybe we have to filter out orders before storing them in a cache.
Might need to be reevaluated, though.

Reply:
Valid concern. The bad news is that we usually receive only 1 order update at once, so we need to lock and unlock for each one. I guess we can try and see.